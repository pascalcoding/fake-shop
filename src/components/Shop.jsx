import NavBar from './NavBar';
import Home from './pages/Home';

function Shop() {
  return (
    <>
      <NavBar />
      <Home />
    </>
  );
}

export default Shop;
