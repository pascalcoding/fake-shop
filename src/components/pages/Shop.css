.products {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
  padding: 15px 50px;
}
