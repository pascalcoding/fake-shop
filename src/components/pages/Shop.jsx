function Shop() {
  return (
    <>
      <div className="products"></div>
    </>
  );
}

export default Shop;
