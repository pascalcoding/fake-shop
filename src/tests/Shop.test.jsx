import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Shop from '../components/pages/Shop';
import { MemoryRouter } from 'react-router-dom';

describe('Shop component', () => {
  it('Component renders correctly', () => {
    const products = [
      {
        id: 1,
        title: 'Test Product 1',
        price: 100,
        category: 'Electronics',
        description: 'Test product 1 is very good',
        image: '',
      },
      {
        id: 2,
        title: 'Test Product 2',
        price: 150,
        category: 'Electronics',
        description: 'Test product 2 is very bad',
        image: '',
      },
    ];

    const { container } = render(
      <MemoryRouter>
        <Shop products={products} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
