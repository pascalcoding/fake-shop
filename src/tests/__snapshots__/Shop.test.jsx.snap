// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop component > Component renders correctly 1`] = `
<div>
  <div
    class="products"
  >
    <a
      class="card"
      href="/Shop/product/1"
    >
      <h3>
        Test Product 1
      </h3>
      <img
        src=""
        style="max-width: 150px; max-height: 100px;"
      />
      <div>
        100
         €
      </div>
    </a>
    <a
      class="card"
      href="/Shop/product/2"
    >
      <h3>
        Test Product 2
      </h3>
      <img
        src=""
        style="max-width: 150px; max-height: 100px;"
      />
      <div>
        150
         €
      </div>
    </a>
  </div>
</div>
`;
